### PR TITLE
Conditionally use local sources for Firestore

### PR DESCRIPTION
**Pull request by wilhuff**
This will force post-merge builds to validate compatibility with pre-release versions of other Firebase pods.

### CODE REVIEW

File: Firestore/Example/Podfile [hunk 1]
```diff
@@ -39,34 +79,20 @@ target 'Firestore_Example_iOS' do
   pod 'Firebase/CoreOnly', '6.6.0'
 
   pod 'FirebaseFirestore', :path => '../../'
+  configure_local_pods()
```
Comment (paulb777) [inline Firestore/Example/Podfile]: I think putting this here breaks the Swift tests

Comment (wilhuff) [reply Firestore/Example/Podfile]: What broke the Swift tests was actually removing the local pod for FirebaseFirestoreSwift. I've added this to `configure_test` and that resolves the build issue.

I've also merged #3647, cleaned this up, and added comments. PTAL.

Note that handling FirebaseFirestoreSwift uniformly resulted in a diff to the project.

**Comment by paulb777** [comment]
Looks like a Swift test problem on the Xcode 8 build

**Comment by wilhuff** [comment]
I've addressed the questionable bits in here and this is now passing.

File: Firestore/Example/Podfile [hunk 2]
```diff
@@ -29,17 +29,50 @@ post_install do |installer|
   end
 end
 
+# Returns true if the user has explicitly requested local sources or if this is
+# a non-PR Travis build.
+def use_local_sources()
+  return ENV.has_key?('USE_LOCAL_SOURCES') || ENV['TRAVIS_PULL_REQUEST'] == 'false'
```
Comment (paulb777) [inline Firestore/Example/Podfile]: Looks like this is not working for cron builds

Comment (wilhuff) [reply Firestore/Example/Podfile]: In this [cron build](https://travis-ci.org/firebase/firebase-ios-sdk/jobs/576121850) in the ` before_install` output I see it working:

```
Fetching podspec for `FirebaseAuth` from `../..`
Fetching podspec for `FirebaseAuthInterop` from `../..`
Fetching podspec for `FirebaseCore` from `../..`
Fetching podspec for `FirebaseCoreDiagnostics` from `../..`
Fetching podspec for `FirebaseCoreDiagnosticsInterop` from `../..`
Fetching podspec for `FirebaseFirestore` from `../../`
Fetching podspec for `FirebaseFirestoreSwift` from `../../`
Fetching podspec for `GoogleBenchmark` from `GoogleBenchmark.podspec`
Fetching podspec for `GoogleDataTransport` from `../..`
Fetching podspec for `GoogleDataTransportCCTSupport` from `../..`
Fetching podspec for `GoogleTest` from `GoogleTest.podspec`
Fetching podspec for `GoogleUtilities` from `../..`
Fetching podspec for `LibFuzzer` from `LibFuzzer.podspec`
Fetching podspec for `ProtobufCpp` from `ProtobufCpp.podspec`
```

Could you point me at an instance where it's not working?

Comment (paulb777) [reply Firestore/Example/Podfile]: https://travis-ci.org/firebase/firebase-ios-sdk/jobs/576121863

Comment (wilhuff) [reply Firestore/Example/Podfile]: Oh, I thought you meant the mechanism for detecting non-PR builds wasn't working, ha.

Yeah, so the problem here is that I'm editing the dependency out of the podspec but not here. I'll send a fix.

Comment (paulb777) [reply Firestore/Example/Podfile]: Sounds good! And on closer inspection of the logs, the Xcode 8 build starting failing on all post-submits after this merge - not just the cron.

**Pull request merged**